model_analyzer/record/metrics_manager.py

```diff
@@ -182,14 +183,19 @@ def execute_run_config(self, run_config):
                 "Existing measurement found for run config. Skipping profile")
             return measurement
 
-        self._server.start(env=run_config.triton_environment())
-        if not self._load_model_variants(run_config):
+        current_model_variants = run_config.model_variants_name()
+        if current_model_variants != self._loaded_models:
             self._server.stop()
-            return
+            self._server.start(env=run_config.triton_environment())
```

Comment:
In the very first call to this method, it seems like `self._server.stop()` gets called before the server has been started. What happens in such case?

Reply:
All stop() calls are gated at the start to make sure there is actually a process. For example:
https://github.com/triton-inference-server/model_analyzer/blob/main/model_analyzer/triton/server/server_local.py#L110